### PR TITLE
Change annotation-based CORS to match configuration-based defaults

### LINKED ISSUE #9504: Configuration- and annotation- based  CORS should have same behavior.
### Issue description

Configuration-based CORS allows all headers and all origins when turned on, unless constrained in a more fine-grained manner using the `allowed-headers` and `allowed-origins` options. Annotation-based CORS should behave exactly the same way.

see #9470

### PR DESCRIPTION
**Pull request by wetted**
Change CrossOrigin annotation-based CORS to match configuration-based CORS defaults so that enabling CORS allows all origins by default. This is a breaking change from Micronaut 3.9.x

closes #9504

### CODE REVIEW

File: src/main/docs/guide/httpServer/serverConfiguration/cors/corsAllowedOrigins.adoc [hunk 1]
```diff
@@ -18,4 +18,4 @@ micronaut:
             - http://foo.com
 ----
 
-WARNING: Use the `allowed-origins-regex` configuration judiciously. You may accidentally make an insecure configuration which could be targeted by an attacker registering domains targeting the regular expression.
+WARNING: Use the `allowed-origins-regex` configuration judiciously. You may accidentally make an insecure configuration which could be targeted by an attacker registering domains targeting the regular expression. Note that if `allowed-origins-regex` is set then `allowed-origins` is ignored.
```
Comment (sdelamo) [inline src/main/docs/guide/httpServer/serverConfiguration/cors/corsAllowedOrigins.adoc]: is that so when you use cors via configuration?

File: src/main/docs/guide/appendix/breaks.adoc [hunk 2]
```diff
@@ -206,3 +206,8 @@ old behavior.
 ==== `OncePerRequestHttpServerFilter` removed
 
 Since Micronaut 3.0 the `OncePerRequestHttpServerFilter` class was deprecated and marked for removal. This class is now removed. Implement link:{api}/io/micronaut/http/filter/HttpServerFilter.html[HttpServerFilter] instead, and replace any usages of `micronaut.once` attributes with a custom attribute name.
+
+==== CORS support with the `@CrossOrigin` annotation
+
+In previous and current versions of Micronaut Framework, configuration-based CORS defaults to allow all origins when CORS is enabled and `allowed-origins` is not set otherwise. However, in previous versions of Micronaut Framework using the `@CrossOrigin` annotation defaults to no origins allowed, requiring the `allowedOrigins` attribute to be set. In Micronaut 4, `@CrossOrigin` is changed to match the behavior of configuration-based CORS. In other words, enabling CORS with `@CrossOrigin` without specifying any attributes means all origins are allowed.
```
Comment (sdelamo) [inline src/main/docs/guide/appendix/breaks.adoc]:
```suggestion
Micronaut Framework 4 changes `@CrossOrigin`  behavior to match configuration-based CORS behavior. A method annotated with `@CrossOrigin` allows any origin if you don't specify any value for the `allowedOrigins` and `allowedOriginsRegex` members.
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by yawkat** [comment]
i dont think we should allow all headers by default

**Comment by sdelamo** [comment]
> i dont think we should allow all headers by default

what default do you think we should use?

**Comment by yawkat** [comment]
no allowed headers

**Comment by sdelamo** [comment]
@yawkat can you review the contents of this PR? I think it would be important to have configuration and annotation based CORs aligned for  MN$4. 

I have moved  the suggestion of changing the default of allowed headers to https://github.com/micronaut-projects/micronaut-core/issues/9518

**Review by yawkat — APPROVED** [review]

File: src/main/docs/guide/httpServer/serverConfiguration/cors/corsAllowedOrigins.adoc (hunk 1 above)
Comment (wetted) [reply src/main/docs/guide/httpServer/serverConfiguration/cors/corsAllowedOrigins.adoc]: I don't think configuration prevents setting both. I'm checking to see what happens. I'll write a test.

Comment (wetted) [reply src/main/docs/guide/httpServer/serverConfiguration/cors/corsAllowedOrigins.adoc]: OK, it looks like with configuration, you use something like

```
micronaut.server.cors.configurations.foo.allowed-origins-regex = ^http(|s):\\/\\/foo\\.com$
micronaut.server.cors.configurations.foo.allowed-origins = https://bar.com
```

then both `http://foo.com` and `https://bar.com` are allowed.

I stand corrected and will change annotation-based to match.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**